main.nf

```diff
@@ -19,24 +20,45 @@ process remove_tech_duplicates {
     """
 }
 
+process build_cell_ontology_dict{
+    publishDir "${params.results_dir}", mode: 'copy'
+    conda "${baseDir}/envs/cell_types_analysis.yaml"
+    input:
+        file(ref_labels_file) from REF_LABELS_FILTERED
```

Comment:
again, rather than using data from config, you should provide `ref_labels_file` as an input to the script